homeassistant/components/emulated_hue.py

```diff
@@ -0,0 +1,538 @@
+        entity = self.hass.states.get(entity_id)
+        if entity is None:
+            return self.Response("Entity not found", status=404)
+
+        if not self.is_entity_exposed(entity):
+            return self.Response("Entity not found", status=404)
+
+        # Parse the request into requested "on" status and brightness
+        parsed = parse_hue_api_put_light_body(request_json, entity)
+
+        if parsed is None:
+            return self.Response("Bad request", status=400)
+
+        result, brightness = parsed
+
+        # Convert the resulting "on" status into the service we need to call
+        service = SERVICE_TURN_ON if result else SERVICE_TURN_OFF
+
+        # Construct what we need to send to the service
+        data = {ATTR_ENTITY_ID: entity_id}
+
+        if brightness is not None:
+            data[ATTR_BRIGHTNESS] = brightness
+
+        if entity.domain.lower() in config.off_maps_to_on_domains:
+            # Map the off command to on
+            service = SERVICE_TURN_ON
+
+            # Caching is required because things like scripts and scenes won't
+            # report as "off" to Alexa if an "off" command is received, because
+            # they'll map to "on". Thus, instead of reporting its actual
+            # status, we report what Alexa will want to see, which is the same
+            # as the actual requested command.
+            self.cached_states[entity_id] = (result, brightness)
+
+        # Perform the requested action
+        self.hass.services.call(core.DOMAIN, service, data, blocking=True)
+
+        json_response = [create_hue_success_response(entity_id, 'on', result)]
+
+        if brightness is not None:
+            json_response.append(
+                create_hue_success_response(entity_id, 'bri', brightness))
+
+        return self.json(json_response)
+
+    def is_entity_exposed(self, entity):
+        """Determine if an entity should be exposed on the emulated bridge."""
+        config = self.config
+
+        if entity.attributes.get('view') is not None:
+            # Ignore entities that are views
+            return False
+
+        domain = entity.domain.lower()
+        explicit_expose = entity.attributes.get(ATTR_EMULATED_HUE, None)
+
+        domain_exposed_by_default = \
+            config.expose_by_default and domain in config.exposed_domains
+
+        # Expose an entity if the entity's domain is exposed by default and
+        # the configuration doesn't explicitly exclude it from being
+        # exposed, or if the entity is explicitly exposed
+        is_default_exposed = \
+            domain_exposed_by_default and explicit_expose is not False
+
+        return is_default_exposed or explicit_expose
+
+
+def parse_hue_api_put_light_body(request_json, entity):
+    """Parse the body of a request to change the state of a light."""
+    if 'on' in request_json:
+        if not isinstance(request_json['on'], bool):
+            return None
+
+        if request_json['on']:

```

Comment:
The 'on' referenced there is from the Hue API, and is independent of Home Assistant's 'on' state. Should I still break it out into a constant?